python/cudf/cudf/core/dataframe.py

```diff
@@ -5874,13 +5874,23 @@ def _from_arrays(
         -------
         DataFrame
         """
-        if data.ndim != 1 and data.ndim != 2:
+        array_data: np.ndarray | cupy.ndarray
+        if hasattr(data, "__cuda_array_interface__"):
+            array_data = cupy.asarray(data, order="F")
```

Comment:
I think the `order` argument is ignored for `data` that is not a `cupy.ndarray` object: https://docs.cupy.dev/en/stable/reference/generated/cupy.asarray.html. If we need to assert memory order, we should use `reshape` as suggested here:
https://github.com/rapidsai/cuspatial/pull/1418#discussion_r1697880062

Reply:
Playing around with this in Colab seems to do the right thing

@seberg what would you recommend we use here (and for the NumPy line below)?

Reply:
I would think this is fine, NumPy will make the copy to get the right order here.  Not sure about CuPy, but I think it would be a bug if it doesn't.
(Using reshape would seem like a weird way to assert memory order, if there is an issue, just call `asarray`/`asfortranarray` a second time.)

In theory, doing it here should be fastest, I think.  NumPy isn't actually good at it (not sure about CuPy), so it could be just as well to do the `.copy()` in the `column.as_column()` call below at the moment (lower peak memory use).
But, I don't really think that matters.

Reply:
Thanks Sebastian! 🙏

Given that will mark this thread resolved